### PR TITLE
eslint configuration

### LINKED ISSUE #20: Configure ESLint Ruleset
- [x] Use https://github.com/tis-gmbh/vscode-migrate/blob/main/.eslintrc.json as base
- [x] Always `;`
- [x] No empty bodies
- [ ] Always typed
- [x] When possible use `const`
- [x] JSX rules
  - [x] No `<Test></Test>` instead `<Test />`

### PR DESCRIPTION
**Pull request by Escowei98**
Pull Request for ESLint configuration

Closes #20

### CODE REVIEW

**Comment by timvahlbrock** [comment]
Kannst wenn du fertig bist gerne mich als Reviewer eintragen.

File: frontend/.eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  "plugins": [
+    "@typescript-eslint",
+    "import"
+  ],
+  "extends": [
+    "plugin:react/recommended",
+    "../.eslintrc.json"
+  ],
+  "rules": {
+    "arrow-body-style": ["error", "as-needed"]
```
Comment (timvahlbrock) [inline frontend/.eslintrc.json:17]: die hier vielleicht auch in die basis?

**Review by timvahlbrock — CHANGES_REQUESTED** [review]

**Comment by timvahlbrock** [comment]
is this PR still a draft?

**Review by timvahlbrock — CHANGES_REQUESTED** [review]
Is it possible, that his configuration does not include "No empty bodies" and "No <Test></Test> instead <Test />" from the issues todos?

**Comment by timvahlbrock** [comment]
> The Rule "no-empty" sets the rule for no empty bodies
> what do you mean by "no instead"?

Sorry, seems like not everything was copied. I meant "no `<Test></Test>` instead `<Test />`". How is that integrated?

**Comment by Escowei98** [comment]
Oh yes you are right, i thought it was included in the "plugin:react/recommended"-rules
I added them to the frontend rules now

File: frontend/package.json
Comment (timvahlbrock) [inline frontend/package.json:1]: One more thing: The lint command in the backend package.json runs lint with --fix while the one in the frontend does not. Can you add it to the frontend command as well?

**Review by timvahlbrock — CHANGES_REQUESTED** [review]

**Comment by timvahlbrock** [comment]
@Escowei98 do you think you can update this PR any time soon? I think this would be very helpful with our upcoming tasks.

**Comment by Escowei98** [comment]
i have pushed both changes (debugger rule and --fix run for frontend)

File: .eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    "plugin:@typescript-eslint/recommended"
+  ],
+  "rules": {
+    "arrow-body-style": ["error", "as-needed"],
+    "no-debugger": 2,
+    "no-console": 0,
+    "no-empty": "error",
+    "@typescript-eslint/explicit-function-return-type": "error",
+    "@typescript-eslint/no-floating-promises": "error",
+    "@typescript-eslint/no-explicit-any": "off",
```
Comment (timvahlbrock) [inline .eslintrc.json:20]: This rule definitely needs to stay on. If you need help finding the correct types, hit me up.

Comment (Escowei98) [reply .eslintrc.json]: turned it back on

Comment (timvahlbrock) [reply .eslintrc.json]: Great thanks. Sorry about that, but this rule is very important to keep the maintainability of the code high.

File: .vscode/settings.json [hunk 3]
```diff
@@ -19,4 +19,9 @@
         "editor.defaultFormatter": "vscode.typescript-language-features",
         "editor.formatOnSave": true
     },
+    "eslint.workingDirectories": [
+        "./app",
```
Comment (timvahlbrock) [inline .vscode/settings.json:23]: What is the ./app for?

**Review by timvahlbrock — APPROVED** [review]

**Comment by timvahlbrock** [comment]
Next time, please address conversations before merging.

**Pull request merged**